### PR TITLE
Sorted Zig-Zag Array Implementation in C++

### PR DESCRIPTION
**Pull request by imsushant12**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

I have added the implementation of "Print Sorted Zig Zag Array" with proper comments and clear explanation. By zig zag array, I mean every element should be either greater than its left and right element or smaller than left or right element.
Example- Input is : 1234 --> Output is :1324

### Checklist

- [ YES ] I've read the contribution guidelines.
- [ YES ] I've checked the issue list before deciding what to submit.
- [ YES ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

https://github.com/TesseractCoding/NeoAlgo/issues/6290

### CODE REVIEW

**Review by ankitaggarwal23 — COMMENTED** [review]
Read the instructions mentioned on Gssoc discord yesterday. Do the changes as per that.

**Comment by imsushant12** [comment]
> Read the instructions mentioned on Gssoc discord yesterday. Do the changes as per that.

But I have written the code as per the instructions can you please specify what have I missed?

File: C-Plus-Plus/other/zig-zag_array_sorted.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/*
+Approach :
+First sort the array then, swap pairs of elements
+except the first element.
+Example - keep a[0], swap a[1] with a[2], swap a[3] with a[4], and so on.
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/other/zig-zag_array_sorted.cpp]: Add a short description

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Already mentioned on server that keep your PR name same as issue name followed by language

File: C-Plus-Plus/other/zig-zag_array_sorted.cpp (hunk 1 above)
Comment (imsushant12) [reply C-Plus-Plus/other/zig-zag_array_sorted.cpp]: Done, please review it.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**